x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/function/FunctionDefinition.java

```diff
@@ -17,28 +17,19 @@
      */
     @FunctionalInterface
     public interface Builder {
-        Function build(UnresolvedFunction uf, boolean distinct, Configuration configuration);
+        Function build(UnresolvedFunction uf, Configuration configuration, Object... extras);
```

Comment:
This is the entry point of this PR - the varargs allow subclasses to specify extra parameters however it's up to them to do routing and verify parameters.

Reply:
Could you maybe add some more details to the javadoc of the Builder?